lib/engine/game/g_18_sj/game.rb

```diff
@@ -788,8 +788,10 @@ def perform_nationalization
             return
           end
 
-          # Merge the corporation with highest share price, and use the first operated as tie break
-          merged = candidates.max_by { |c| [c.share_price.price, -@round.entities.find_index(c)] }
+          # Merge the corporation with highest share price, and use highest token on the pile as tie break
+          merged = candidates.max_by do |c|
+            [c.share_price.price, c.share_price.coordinates.last, c.share_price.corporations.find_index(c)]
```

Comment:
The share_price corporations array is ordered such that the token on the top of the stack is first, with the index increasing as you move down the pile of tokens. So you'd need to use `-share_price.corporations.index(c)` here, otherwise ties will be broken in favour of the tokens lower down the stack.

And I've checked the code for the default sorting of corporations, and it's the opposite of what I thought. So `candiates.min` should work for you.
https://github.com/tobymao/18xx/blob/011ed24d0e169c1f9a203105fe9078561591a0e1/lib/engine/corporation.rb#L93-L106

Reply:
yep that works too. Done.